src/main/java/teammates/googleSendgridJava/Sendgrid.java

```diff
@@ -0,0 +1,755 @@
+                JSONObject apiResponse = new JSONObject(response);
+                JSONArray errorsObj = (JSONArray) apiResponse.get("errors");
+                for (int i = 0; i < errorsObj.length(); i++) {
+                    if (i != 0) {
+                        serverResponse += ", ";
+                    }
+                    serverResponse += errorsObj.get(i);
+                }
+                w.warning(serverResponse, null);
+            }
+        } catch (MalformedURLException e) {
+            w.warning("Malformed URL Exception", e);
+        } catch (IOException e) {
+            w.warning("IO Exception", e);
+        }
+    }
+
+    /**
+     * _addToName - Append an recipient name to the existing list of names
+     *
+     * @param    email   Recipient email address
+     * @param    name    Recipient name
+     * @return           The SendGrid object.
+     */
+    private Sendgrid _addToName(String name) {
+        this.to_name_list.add(name);
+
+        return this;
+    }
+
+    /**
+     * useHeaders - Checks to see whether or not we can or should you headers. In most cases,
+     * we prefer to send our recipients through the headers, but in some cases,
+     * we actually don't want to. However, there are certain circumstances in
+     * which we have to.
+     */
+    private Boolean _useHeaders() {
+        if ((this._preferNotToUseHeaders() == true) && (this._isHeadersRequired() == false)) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * _preferNotToUseHeaders - There are certain cases in which headers are not a preferred choice
+     * to send email, as it limits some basic email functionality. Here, we
+     * check for any of those rules, and add them in to decide whether or
+     * not to use headers
+     *
+     * @return if true we don't
+     */
+    private Boolean _preferNotToUseHeaders() {
+        if (this.getBccs().size() == 0)
+        {
+            return true;
+        }
+        if (this.use_headers != null && this.use_headers == false)
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * isHeaderRequired - determines whether or not we need to force recipients through the smtpapi headers
+     *
+     * @return if true headers are required
+     */
+    private Boolean _isHeadersRequired() {
+        if (this.use_headers == true)
+        {
+            return true;
+        }
+
+        return false;
+    }
+    
+    private String _escapeUnicode(String input) {
+        StringBuilder sb = new StringBuilder();
+        int len = input.length();
+        for (int i = 0; i < len; i++) {
+          int code = Character.codePointAt(input, i);
+          if (code > 127) {
+            sb.append(String.format("\\u%x", code));
+          } else {
+            sb.append(String.format("%c", code));
+          }
+        }
+        return sb.toString();
+      }
+}
```

Comment:
Did you get this file from somewhere else? If so, mention the source. Don't assume the code will fit our context out of the box. Sample code sometimes take shortcuts to make things simpler.